### PR TITLE
Add topics from wiki

### PR DESCRIPTION
**Pull request by mhegazy**
Move the missing topics from the wiki to the handbook.

### CODE REVIEW

**Comment by mhegazy** [comment]
@DanielRosenwasser can you take a look

File: pages/Compiler Options.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+`--sourceMap` | | Generates corresponding '.map' file.
+`--sourceRoot` | | Specifies the location where debugger should locate TypeScript files instead of source locations. Use this flag if the sources will be located at run-time in a different location than that at design-time. The location specified will be embedded in the sourceMap to direct the debugger where the source files where be located.
+`--stripInternal`<sup>[1]</sup> | | Do not emit declarations for code that has an `/** @internal */` JSDoc annotation.
+`--suppressImplicitAnyIndexErrors` | | Suppress `--noImplicitAny` errors for indexing objects lacking index signatures. See [issue #1232](https://github.com/Microsoft/TypeScript/issues/1232#issuecomment-64510362) for more details.
+
+<sup>[1]</sup> These options are experimental.
+
+## Related
+
+* For tsconfig.json see [[tsconfig.json]]
```
Comment (DanielRosenwasser) [inline pages/Compiler Options.md]: This will need to be changed to `[tsconfig.json](./tsconfig.json.html)`

File: pages/Compiler Options.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+`--removeComments` | | Remove all comments except copy-right header comments beginning with `/!*`
+`--rootDir` | | Specifies the root directory of input files. Only use to control the output directory structure with `--outDir`.
+`--sourceMap` | | Generates corresponding '.map' file.
+`--sourceRoot` | | Specifies the location where debugger should locate TypeScript files instead of source locations. Use this flag if the sources will be located at run-time in a different location than that at design-time. The location specified will be embedded in the sourceMap to direct the debugger where the source files where be located.
+`--stripInternal`<sup>[1]</sup> | | Do not emit declarations for code that has an `/** @internal */` JSDoc annotation.
+`--suppressImplicitAnyIndexErrors` | | Suppress `--noImplicitAny` errors for indexing objects lacking index signatures. See [issue #1232](https://github.com/Microsoft/TypeScript/issues/1232#issuecomment-64510362) for more details.
```
Comment (DanielRosenwasser) [inline pages/Compiler Options.md]: Do we still want this link?

File: pages/Compiler Options.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+`--listFiles` | | Print names of files part of the compilation.
+`--locale` | | The locale to use to show error messages, e.g. en-us.
+`--mapRoot` | | Specifies the location where debugger should locate map files instead of generated locations. Use this flag if the .map files will be located at run-time in a different location than than the .js files. The location specified will be embedded in the sourceMap to direct the debugger where the map files where be located.
+`--moduleResolution`<sup>[1]</sup> | | Determine how modules get resolved. Either 'node' for Node.js/io.js style resolution, or 'classic' (default).
+`--newLine` | | Use the specified end of line sequence to be used when emitting files: 'CRLF' (dos) or 'LF' (unix)."
+`--noEmit` | | Do not emit outputs.
+`--noEmitOnError` | | Do not emit outputs if any errors were reported.
+`--noEmitHelpers` | | Do not generate custom helper functions like `__extends` in compiled output.
+`--noImplicitAny` | | Raise error on expressions and declarations with an implied 'any' type.
+`--noLib` | | Do not include the default library file (lib.d.ts).
```
Comment (DanielRosenwasser) [inline pages/Compiler Options.md]: Use literal text around `lib.d.ts`

File: pages/Compiler Options.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+`--watch` | `-w` | Run the compiler in watch mode. Watch input files and trigger recompilation on changes.
+`--charset` | | The character set of the input files.
+`--diagnostics` | | Show diagnostic information.
+`--emitBOM` | | Emit a UTF-8 Byte Order Mark (BOM) in the beginning of output files.
+`--emitDecoratorMetadata`<sup>[1]</sup> | | Emit design-type metadata for decorated declarations in source. See [issue #2577](https://github.com/Microsoft/TypeScript/issues/2577) for details.
+`--experimentalAsyncFunctions` | | Support ES7-proposed asynchronous functions using the `async`/`await` keywords.
+`--inlineSourceMap` | | Emit a single file with source maps instead of having a separate file.
+`--inlineSources` | | Emit the source alongside the sourcemaps within a single file; requires `--inlineSourceMap` to be set.
+`--isolatedModules` | | Unconditionally emit imports for unresolved files.
+`--jsx` | | Support JSX in '.tsx' files: 'React' or 'Preserve'. See [[JSX]].
```
Comment (DanielRosenwasser) [inline pages/Compiler Options.md]: `[JSX](./JSX.html)`

File: pages/Compiler Options.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+`--mapRoot` | | Specifies the location where debugger should locate map files instead of generated locations. Use this flag if the .map files will be located at run-time in a different location than than the .js files. The location specified will be embedded in the sourceMap to direct the debugger where the map files where be located.
+`--moduleResolution`<sup>[1]</sup> | | Determine how modules get resolved. Either 'node' for Node.js/io.js style resolution, or 'classic' (default).
+`--newLine` | | Use the specified end of line sequence to be used when emitting files: 'CRLF' (dos) or 'LF' (unix)."
+`--noEmit` | | Do not emit outputs.
+`--noEmitOnError` | | Do not emit outputs if any errors were reported.
+`--noEmitHelpers` | | Do not generate custom helper functions like `__extends` in compiled output.
+`--noImplicitAny` | | Raise error on expressions and declarations with an implied 'any' type.
+`--noLib` | | Do not include the default library file (lib.d.ts).
+`--noResolve` | | Do not add triple-slash references or module import targets to the list of compiled files.
+`--out` | | DEPRECATED. Use `--outFile` instead.
```
Comment (DanielRosenwasser) [inline pages/Compiler Options.md]: Just remove this.

File: pages/tsconfig.json.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        "exclude": [
+            "node_modules",
+            "wwwroot"
+        ]
+    }
+    ```
+
+## Details
+
+The `"compilerOptions"` property can be omitted, in which case the compiler's defaults are used. See our full list of supported [[Compiler Options]].
```
Comment (DanielRosenwasser) [inline pages/tsconfig.json.md]: `[Compiler Options](./compiler-options.html)`

File: pages/Compiler Options.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
+## Compiler Options
+
+Option | Shorthand | Description
+--------|-------|--------------
+`--declaration` | `-d` | Generates corresponding '.d.ts' file.
+`--help` | `-h` | Print help message.
+`--version` | `-v` | Print the compiler's version.
+`--module` | `-m` | Specify module code generation: 'commonjs', 'amd', 'system', or 'umd'. Only 'amd' and 'system' can be used in conjunction with `--outFile`.
+`--project` | `-p` | Compile the project in the given directory. The directory needs to contain a `tsconfig.json` file to direct compilation. See [[tsconfig.json]] documentation for more details.
```
Comment (DanielRosenwasser) [inline pages/Compiler Options.md]: `[tsconfig.json](./tsconfig.json.html)`

File: pages/Compiler Options.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+`--sourceRoot` | | Specifies the location where debugger should locate TypeScript files instead of source locations. Use this flag if the sources will be located at run-time in a different location than that at design-time. The location specified will be embedded in the sourceMap to direct the debugger where the source files where be located.
+`--stripInternal`<sup>[1]</sup> | | Do not emit declarations for code that has an `/** @internal */` JSDoc annotation.
+`--suppressImplicitAnyIndexErrors` | | Suppress `--noImplicitAny` errors for indexing objects lacking index signatures. See [issue #1232](https://github.com/Microsoft/TypeScript/issues/1232#issuecomment-64510362) for more details.
+
+<sup>[1]</sup> These options are experimental.
+
+## Related
+
+* For tsconfig.json see [[tsconfig.json]]
+* For Setting the compiler options in MSBuild projects see [[Setting Compiler Options in MSBuild projects]]
```
Comment (DanielRosenwasser) [inline pages/Compiler Options.md]: [Setting Compiler Options in MSBuild projects](./setting-compiler-options-in-msbuild-projects.html)

**Comment by mhegazy** [comment]
@DanielRosenwasser Ping!

File: pages/Integrating with Build Tools.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+
+var out = path.join(__dirname, "output.js")
+
+Duo(__dirname)
+  .entry('entry.ts')
+  .use(typescript())
+  .run(function (err, results) {
+    if (err) throw err;
+    // Write compiled result to output file
+    fs.writeFileSync(out, results.code);
```
Comment (DanielRosenwasser) [inline pages/Integrating with Build Tools.md]: Can you standardize the indentation for these examples?

File: pages/Compiler Options in MSBuild.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+```XML
+  <PropertyGroup Condition="'$(Configuration)' == 'Debug'">
+    <TypeScriptRemoveComments>false</TypeScriptRemoveComments>
+    <TypeScriptSourceMap>true</TypeScriptSourceMap>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)' == 'Release'">
+    <TypeScriptRemoveComments>true</TypeScriptRemoveComments>
+    <TypeScriptSourceMap>false</TypeScriptSourceMap>
+  </PropertyGroup>
+  <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\TypeScript\Microsoft.TypeScript.targets" Condition="Exists('$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\TypeScript\Microsoft.TypeScript.targets')" />
```
Comment (DanielRosenwasser) [inline pages/Compiler Options in MSBuild.md]: This is absurdly long. Can you at least split it into 4 lines?

**Comment by DanielRosenwasser** [comment]
@mhegazy can you change the names to the appropriate markdown file paths as per #114?

**Comment by mhegazy** [comment]
:bell:

File: pages/Typings for NPM Packages.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        "main": "./lib/foo.js",
+        "typings": "./lib/foo.d.ts"
+    }
+    ```
+
+2. Try to load a file named `index.d.ts` located in the package folder (`node_modules/foo/`) - this file should contain typings for the package.
+
+The precise algorithm for module resolution can be found [here](https://github.com/Microsoft/TypeScript/issues/2338)
+
+### What your typings file should
```
Comment (DanielRosenwasser) [inline pages/Typings for NPM Packages.md]: What my typings file should

File: pages/tsconfig.json.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
+## Overview
+
+The presence of a `tsconfig.json` file in a directory indicates that the directory is the root of a TypeScript project.
+The `tsconfig.json` file specifies the root files and the compiler options required to compile the project.
+`tsconfig.json` is supported starting with TypeScript 1.5.
```
Comment (DanielRosenwasser) [inline pages/tsconfig.json.md]: Remove the 1.5 thing, it's a 1.7 world now

File: pages/Compiler Options in MSBuild.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+`--preserveConstEnums`                       | TypeScriptPreserveConstEnums               | boolean
+`--removeComments`                           | TypeScriptRemoveComments                   | boolean
+`--rootDir`                                  | TypeScriptRootDir                          | File path
+`--isolatedModules`                          | TypeScriptIsolatedModules                  | boolean
+`--sourceMap`                                | TypeScriptSourceMap                        | File path
+`--sourceRoot`                               | TypeScriptSourceRoot                       | File path
+`--suppressImplicitAnyIndexErrors`           | TypeScriptSuppressImplicitAnyIndexErrors   | boolean
+`--suppressExcessPropertyErrors`             |  TypeScriptSuppressExcessPropertyErrors    | boolean
+`--moduleResolution`<sup>[1]</sup>           | TypeScriptModuleResolution                 | `Classic` or `NodeJs`
+`--experimentalAsyncFunctions`<sup>[1]</sup> | TypeScriptExperimentalAsyncFunctions       | boolean
```
Comment (DanielRosenwasser) [inline pages/Compiler Options in MSBuild.md]: This can be removed.

File: pages/Compiler Options in MSBuild.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+`--jsx`<sup>[1]</sup>                        | TypeScriptJSXEmit                          | `React` or `Preserve`
+`--project`                                  | *Not supported in VS*                      |
+`--watch`                                    | *Not supported in VS*                      |
+`--diagnostics`                              | *Not supported in VS*                      |
+`--listFiles`                                | *Not supported in VS*                      |
+`--noEmit`                                   | *Not supported in VS*                      |
+*VS only option*                             | TypeScriptAdditionalFlags                  | *Any compiler option*
+
+<sup>1</sup> These settings are new in TypeScript 1.6
```
Comment (DanielRosenwasser) [inline pages/Compiler Options in MSBuild.md]: Remove this and all references to this

File: pages/tsconfig.json.md [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+            "wwwroot"
+        ]
+    }
+    ```
+
+## Details
+
+The `"compilerOptions"` property can be omitted, in which case the compiler's defaults are used. See our full list of supported [Compiler Options](./Compiler Options.md).
+
+If no `"files"` property is present in a `tsconfig.json`, the compiler defaults to including all TypeScript (\*.ts or \*.tsx) files in the containing directory and subdirectories.
```
Comment (DanielRosenwasser) [inline pages/tsconfig.json.md]: Use literal text here,

File: pages/tsconfig.json.md [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+When a `"files"` property is present, only the specified files are included.
+
+If the `"exclude"` property is specified, the compiler includes all TypeScript (\*.ts or \*.tsx) files in the containing directory and subdirectories except for those files or folders that are excluded.
```
Comment (DanielRosenwasser) [inline pages/tsconfig.json.md]: Use literal text for `*.ts` and `.tsx`

File: pages/Typings for NPM Packages.md [hunk 17]
```diff
@@ -0,0 +1,26 @@
+TypeScript 1.6 has introduced new way of resolving module names that mimics the Node.js module resolution algorithm. This means the TypeScript compiler can currently load typings that are bundled with npm packages. The compiler will try to discover typings for module `"foo"` using the following set of rules:
+
+1. Try to load the `package.json` file located in the appropriate package folder (`node_modules/foo/`). If present,read the path to the typings file described in the `"typings"` field. For example, in the following `package.json`, the compiler will resolve the typings at `node_modules/foo/lib/foo.d.ts`
+
+    ```JSON
+    {
+        "name": "foo",
```
Comment (DanielRosenwasser) [inline pages/Typings for NPM Packages.md]: Consider LaTeX-importer

File: pages/Typings for NPM Packages.md [hunk 18]
```diff
@@ -0,0 +1,26 @@
+TypeScript 1.6 has introduced new way of resolving module names that mimics the Node.js module resolution algorithm. This means the TypeScript compiler can currently load typings that are bundled with npm packages. The compiler will try to discover typings for module `"foo"` using the following set of rules:
```
Comment (DanielRosenwasser) [inline pages/Typings for NPM Packages.md]: Consider removing 1.6

Comment (DanielRosenwasser) [inline pages/Typings for NPM Packages.md]: TypeScript supplies a way of resolving module names that mimics the Node.js module resolution algorithm.

Comment (DanielRosenwasser) [inline pages/Typings for NPM Packages.md]: Split this into 3 lines

File: pages/Compiler Options in MSBuild.md [hunk 19]
```diff
@@ -0,0 +1,66 @@
...
+
+## What is supported in my version of Visual Studio
```
Comment (DanielRosenwasser) [inline pages/Compiler Options in MSBuild.md]: question mark at the end

Comment (mhegazy) [inline pages/Compiler Options in MSBuild.md]: linter does not like that

**Pull request merged**